pkg/stargate/deployments.go

```diff
@@ -312,12 +312,11 @@ func computeJvmOptions(template *api.StargateTemplate) string {
 	heapSize := computeHeapSize(template)
 	heapSizeInBytes := heapSize.Value()
 	jvmOptions := fmt.Sprintf("-XX:+CrashOnOutOfMemoryError -Xms%v -Xmx%v", heapSizeInBytes, heapSizeInBytes)
-	if template.CassandraConfigMapRef != nil {
-		jvmOptions += fmt.Sprintf(
-			" -Dstargate.unsafe.cassandra_config_path=%s",
-			cassandraConfigPath,
-		)
-	}
+	jvmOptions += fmt.Sprintf(
+		" -Dstargate.unsafe.cassandra_config_path=%s",
+		cassandraConfigPath,
+	)
+
```

Comment:
If I understand correctly, there is always a config (copied from the Cassandra one), even if the user did not provide their own?
Maybe it would be worth a quick explanation in a comment, with a reference to `stargate_controller.go`.

Reply:
There is always a ConfigMap which can be empty. We are only copying the `server_encryption_options` and `client_encryption_options` properties from Cassandra when they are set.

+1 to adding a comment.

Reply:
Should we add/update a test to verify that the JVM property is always set and that the ConfigMap is always created?

Reply:
> Should we add/update a test to verify that the JVM property is always set and that the ConfigMap is always created?

I did so by modifying the single DC Stargate encryption test. It was passing before the fix because a `cassandra-config` config map was created before the K8ssandraCluster was created. Otherwise it would've been failing.
I've removed that config map to verify that the generated one would be passed on to Stargate correctly.

Reply:
> We are only copying the `server_encryption_options` and `client_encryption_options` properties from Cassandra when they are set.

That's very useful for #722, thanks.